float32 length
float32 width
int32 size
float32[] point_and_tangent
string track_layout
